modules/axruntime/src/lib.rs

```diff
@@ -186,6 +186,11 @@ pub extern "C" fn rust_main(cpu_id: usize, dtb: usize) -> ! {
     info!("Primary CPU {} init OK.", cpu_id);
     INITED_CPUS.fetch_add(1, Ordering::Relaxed);
 
+    #[cfg(feature = "alloc")]
+    for init in axns::INIT_RESOURCE.iter() {
```

Comment:
We can provide a more general initialization method that is not limited to `axns`, like the `init_array` in unikraft: https://github.com/unikraft/unikraft/blob/0562d8f1c9857619586fb7901e2c54c056121875/lib/ukboot/boot.c#L472

Reply:
It has been published as [ctor_bare](https://crates.io/crates/ctor_bare) crate.